src/EnergyPlus/WaterManager.cc

```diff
@@ -1049,9 +1049,14 @@ namespace WaterManager {
 
         // is tank lower than float valve on capacity and requesting fill from controlled supplier?
         FillVolRequest = 0.0;
-        if ((VolumePredict) < state.dataWaterData->WaterStorage(TankNum).ValveOnCapacity) { // turn on supply to fill tank
+
+        if ( ((VolumePredict) < state.dataWaterData->WaterStorage(TankNum).ValveOnCapacity) ||
+              state.dataWaterData->WaterStorage(TankNum).LastTimeStepFilling )
+        { // turn on supply to fill tank
             FillVolRequest = state.dataWaterData->WaterStorage(TankNum).ValveOffCapacity - VolumePredict;
 
+            state.dataWaterData->WaterStorage(TankNum).LastTimeStepFilling = true;
```

Comment:
If you go below `ValveOnCapacity`, you start filling, and you set the `LastTimeStepFilling` to true.
**Or** if `LastTimeStepFilling` is already `true`, you keep filling.